typescript.md

```diff
@@ -0,0 +1,82 @@
+# Usage with Typescript
+You can import the library and use it with Typescript:
+
+```ts
+import HyperHTMLElement from "hyperhtml-element";
+
+class MySimpleElement extends HyperHTMLElement {
+   render() {
+      this.html`Hello from my-simple-element!`
+   }
+}
+
+MySimpleElement.define('my-simple-element');
+```
+
+You can also define the type of the state using an interface, for have a better auto-complete in your editor. You should also define the observedAttributes that are automaticaly mapped as class attributes. Here's a more complex example:
+
+```ts
+import HyperHTMLElement from "hyperhtml-element";
+
+interface MyCustomElementState {
+   counter: number;
+   name: string;
+}
+
+class MyCustomElement extends HyperHTMLElement<MyCustomElementState> {
+
+   // Observed attributes are mapped to class attributes, converted from kebab-case to camelCase
+   // They should be declared for later use and declared as optional since the user is not forced to use them
+   name?: string;
+
+   static get observedAttributes() {
+      return ['name'];
+   }
+
+   attributeChangedCallback(attrName: string, prev: string, curr: string) {
+      if (attrName === 'name') {
```

Comment:
if you observe `name` there's no way this method would be called with any other attribute, is that `if` needed?

Reply:
In this case it is not needed, you are right, but I put that `if` just to show a "pattern" that can be used when there are multiple attributes. It can be removed anyway.

Reply:
fair enough.